common/src/main/java/com/wynntils/models/label/LabelModel.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.label;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Model;
+import com.wynntils.handlers.labels.event.EntityLabelChangedEvent;
+import com.wynntils.models.label.event.LabelIdentifiedEvent;
+import com.wynntils.models.label.parsers.GatheringNodeLabelParser;
+import com.wynntils.models.label.parsers.GatheringStationLabelParser;
+import com.wynntils.models.label.parsers.NpcLabelParser;
+import com.wynntils.models.label.type.LabelInfo;
+import com.wynntils.models.label.type.LabelParser;
+import com.wynntils.utils.mc.type.Location;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class LabelModel extends Model {
+    private final List<LabelParser> parsers = new ArrayList<>();
+
+    public LabelModel() {
+        super(List.of());
+
+        registerParsers();
+    }
+
+    @SubscribeEvent
+    public void onLabelChange(EntityLabelChangedEvent event) {
+        for (LabelParser parser : parsers) {
+            LabelInfo info = parser.getInfo(
+                    event.getName(), Location.containing(event.getEntity().position()));
+
+            if (info == null) continue;
+
+            WynntilsMod.postEvent(new LabelIdentifiedEvent(info));
+            return;
+        }
+    }
+
+    private void registerParsers() {
+        // Special order for optimization
+        registerParser(new NpcLabelParser());
```

Comment:
This kind of feels a bit more like a Handler than a Model. Wynn uses entity labels for a variety of purposes, so the label itself does not really match to something we want to model as part of the world, but rather as a way we get info that we want to model.

If we need to have a certain order for the parsers it will make the inverse dependency registration a bit more tricky, but we can probably have like `Handlers.Label.registerParser(new NpcLabelParser, HIGH_PRIORITY)` or something like that.

Reply:
We have LabelHandler. But it handles generic entity name tag change events.

I opted for a LabelModel since we do wynn specific parsing here.

Reply:
Actually, we already *have* a LabelHandler. (I was a bit surprised, since I thought I had added one).

I think this functionality should be folded into that.

Reply:
We can do "wynn specific" parsing in a handler, as long as we have the dependency inversion pattern of e.g. ActionBarHandler, where we register wynn-specific callbacks.

I think this would be better redesigned so that LabelHandler has the additional responsibility of sending new labels with a list of registered parsers, and sending out a LabelIdentifiedEvent with a LabelInfo, when a parser returns non-null.

The actual parsers should reside in more specific models. The gathering stuff clearly belongs to ProfessionModel. We do not really have a NpcModel, something I have been pondering for a while. (I would like to connect NPC characters with NPC dialogue lines, and have a place to prepare for Voices of Wynn integration.) So maybe it is time to create one.

Reply:
One way of thinking about this is that I believe a "model" should match a concept that any Wynncraft player will have. They will know stuff such as emeralds, rewards, damage, profession. Stuff like "label" is very much minecraft-centric. I ljust ooked through all our models, and there is one more that really should be a handler as well, and that is particle. It is the same situation -- particles are a minecraft concept; we need to separate them for all the different meanings they have in Wynncraft. The user model is that "this kind of particle effects mean it is a lootrun thingie, this other means it is an important quest place, etc".